optionals/compat_sog/functions/fnc_handlePunjiTrapDamage.sqf

```diff
@@ -19,7 +19,7 @@ params ["_trap"];
 if (isNull (configFile >> "CfgPatches" >> "ace_medical")) exitWith {};
 
 private _radius = getNumber (configOf _trap >> "indirectHitRange");
-private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x};
+private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x && isDamageAllowed _x};
```

Comment:
Maybe "lazy select", i.e. 
```suggestion
private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x} select {isDamageAllowed _x};
```

Reply:
Oh cool, I never thought of doing it like that